src/dscacheutil.ts

```diff
@@ -0,0 +1,128 @@
+import { keyValue } from "@fig/autocomplete-generators";
+
+const categoryKeysMapping = {
+  group: ["name", "gid"],
+  host: ["name", "ip_address"],
+  mount: ["name"],
+  protocol: ["name", "number"],
+  rpc: ["name", "number"],
+  service: ["name", "port"],
+  user: ["name", "uid"],
+};
+const categories = Array.from(Object.keys(categoryKeysMapping));
+
+interface PostProcessQuery {
+  attributeKey: string;
+}
+
+const postProcessQuery =
+  (options: PostProcessQuery): Fig.Generator["postProcess"] =>
+  (out) => {
+    const { attributeKey } = options;
+    const values = new Set<string>();
+    out.split("\n\n").map((entry) => {
+      entry.split("\n").forEach((line) => {
+        const [key, value] = line.trim().split(": ");
+        if (key === attributeKey) {
+          values.add(value);
+        }
+      });
+    });
+
+    const suggestions = [];
+    values.forEach((value) => suggestions.push({ name: value }));
+
+    return suggestions;
+  };
+
+const dscacheutilGenerators: Record<string, Fig.Generator> = {
+  keys: {
+    custom: async (tokens, executeShellCommand) => {
+      const category = tokens[tokens.length - 2];
+      return categoryKeysMapping[category].map((key: string) => ({
+        name: key,
+      }));
+    },
+  },
+  values: {
+    custom: async (tokens, executeShellCommand) => {
+      const category = tokens[tokens.length - 4];
+      const attributeKey = tokens[tokens.length - 2];
+      const pp = postProcessQuery({ attributeKey });
+      return pp(
+        await executeShellCommand(`dscacheutil -q ${category}`),
+        tokens
+      );
+    },
+  },
+};
+
+const completionSpec: Fig.Spec = {
+  name: "dscacheutil",
+  description: "Utility for managing the Directory Service cache",
+  subcommands: [
```

Comment:
Single-letter options are behaving strangely because of the chaining rules, if you make this change it should be a lot more intuitive!
```suggestion
const completionSpec: Fig.Spec = {
  name: "dscacheutil",
  description: "Utility for managing the Directory Service cache",
  parserDirectives: {
    flagsArePosixNoncompliant: true,
  },
  subcommands: [
```

Reply:
Including this change seems to make the `dscacheutil -q user -a uid` command break. The `dscacheutilGenerators.keys` doesn't seem to be called

Reply:
@SeparateRecords ^